tests/integration/test_charm.py

```diff
@@ -25,12 +35,60 @@ async def test_build_and_deploy(ops_test: OpsTest):
     """
     # Build and deploy charm from local source folder
     charm = await ops_test.build_charm(".")
-    resources = {RESOURCE_NAME: METADATA["resources"][RESOURCE_NAME]["upstream-source"]}
-
-    # Deploy the charm and wait for active/idle status
-    await asyncio.gather(
-        ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME),
-        ops_test.model.wait_for_idle(
-            apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000
-        ),
+    resources = {RESOURCE_NAME: RESOURCE_PATH}
+
+    # Deploy the charm
+    await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME)
+
+    # and wait for active/idle status
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000
+    )
+
+    # Deploying grafana-agent-k8s and add the logging relation
+    await deploy_and_assert_grafana_agent(
+        ops_test.model, APP_NAME, metrics=False, dashboard=True, logging=True
+    )
+    logger.info(
+        "Adding relation: %s:%s and %s:%s",
+        APP_NAME,
+        APP_GRAFANA_DASHBOARD_DEVICES,
+        GRAFANA_AGENT_APP,
+        GRAFANA_AGENT_GRAFANA_DASHBOARD,
+    )
+    await ops_test.model.integrate(
+        f"{APP_NAME}:{APP_GRAFANA_DASHBOARD_DEVICES}",
+        f"{GRAFANA_AGENT_APP}:{GRAFANA_AGENT_GRAFANA_DASHBOARD}",
```

Comment:
Why would you connect the `grafana-dashboard-device` (that is supposed to push device dashboard to grafana) to the grafana-agent? It should push to grafana-k8s no?

Reply:
The grafana agent is merely a proxy (a mock) of the COS stack here. Since it offer the relations we expect for logging and dashboard, we deploy that instead of the whole COS.